### PR TITLE
UI fixes

### PR DESCRIPTION
**Pull request by palant-dev**
- Fixed the issue with drag gesture being too fast or too slow 
- Refactored code for better readability

### CODE REVIEW

**Review by yukim17 — APPROVED** [review]

File: RayD/RayD/UI/MainView/MainViewModel.swift [hunk 1]
```diff
@@ -164,12 +160,12 @@ final class MainViewModel: ObservableObject {
...
-                    self.state = .stop
-                    return
-                }
-                self.setupTimer(with: uvIndex)
+//                guard uvIndex != 0 else {
+//                    self.error = .zeroUVIndex
+//                    self.state = .stop
+//                    return
+//                }
+                self.setupTimer(with: uvIndex + 1)
```
Comment (GabriellAnnunziata) [inline RayD/RayD/UI/MainView/MainViewModel.swift:168]: Shouldn't this part be uncommented?

File: RayD/RayD/UI/MainView/ContentView.swift [hunk 2]
```diff
@@ -32,8 +30,16 @@ struct ContentView: View {
                 )
                 .overlay {
                     ZStack {
+                        CirularProgressView(
+                            progress: $viewModel.progress,
+                            secondsToCompletion: $viewModel.secondsToCompletion,
+                            scaleOfView: 1.2
```
Comment (yukim17) [inline RayD/RayD/UI/MainView/ContentView.swift:36]: Does it work for different screen sizes?

File: RayD/RayD/UI/MainView/MainViewModel.swift [hunk 3]
```diff
@@ -160,12 +160,12 @@ final class MainViewModel: ObservableObject {
 
                 self.calculatingExposureTime = false
 
-                guard uvIndex != 0 else {
-                    self.error = .zeroUVIndex
-                    self.state = .stop
-                    return
-                }
-                self.setupTimer(with: uvIndex)
+//                guard uvIndex != 0 else {
```
Comment (yukim17) [inline RayD/RayD/UI/MainView/MainViewModel.swift:163]: Please, don't commit this. This hack is only for testing purposes :)

**Review by yukim17 — CHANGES_REQUESTED** [review]

File: RayD/RayD/UI/MainView/ContentView.swift [hunk 4]
```diff
@@ -12,147 +12,191 @@ struct ContentView: View {
...
+            Image("dotsmenu")
+                .resizable()
+                .scaledToFit()
+                .frame(width: 24)
+
+            Spacer()
+
+            Spacer()
+                .frame(width: 24)
+
```
Comment (GabriellAnnunziata) [inline RayD/RayD/UI/MainView/ContentView.swift:103]: Maybe we can try to find something better for the positioning rather then using the frame for the Spacer

**Review by GabriellAnnunziata — APPROVED** [review]

File: RayD/RayD/UI/MainView/ContentView.swift (hunk 2 above)
Comment (palant-dev) [reply RayD/RayD/UI/MainView/ContentView.swift]: agreed on changing later

File: RayD/RayD/UI/MainView/MainViewModel.swift (hunk 3 above)
Comment (palant-dev) [reply RayD/RayD/UI/MainView/MainViewModel.swift:163]: done

File: RayD/RayD/UI/MainView/MainViewModel.swift (hunk 1 above)
Comment (palant-dev) [reply RayD/RayD/UI/MainView/MainViewModel.swift:168]: done

**Pull request merged**